contracts/contracts/coordination/Coordinator.sol

```diff
@@ -124,6 +126,12 @@ contract Coordinator is AccessControlDefaultAdminRules {
         _setRoleAdmin(INITIATOR_ROLE, bytes32(0));
     }
 
+    function setProviderPublicKey(bytes calldata publicKey) external {
+        // TODO: Verify public key length
+        require(publicKey.length == 48, "Invalid public key length");
+        providerPublicKey[msg.sender] = publicKey;
```

Comment:
This should be a mapping from providers to keys, not from operators. You can get the provider address this way:
```js
address provider = application.stakingProviderFromOperator(msg.sender);
```

Reply:
:heavy_check_mark: